Source/d_deh.c

```diff
@@ -1682,15 +1704,30 @@ void ProcessDehFile(const char *filename, char *outfilename, int lumpnum)
           continue;
         }
 
-      for (i=0; i<DEH_BLOCKMAX; i++)
+      for (match=0, i=0; i<DEH_BLOCKMAX; i++)
         if (!strncasecmp(inbuffer,deh_blocks[i].key,strlen(deh_blocks[i].key)))
           { // matches one
-            if (fileout)
-              fprintf(fileout,"Processing function [%d] for %s\n",
-                      i, deh_blocks[i].key);
-            deh_blocks[i].fptr(filein,fileout,inbuffer);  // call function
+            if (i < DEH_BLOCKMAX-1)
+              match = 1;
             break;  // we got one, that's enough for this block
           }
+
+      if (match) // inbuffer matches a valid block code name
+        last_i = i;
+      else
+        { // process that same line again with the last valid block code handler
+          i = last_i;
+          if (!filein->lump)
```

Comment:
Please remove the `if (!filein->lump)` check, `dehfseek()` has a solution for both.